kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/arbs.kt

```diff
@@ -28,6 +31,17 @@ fun <A> Arb<A>.next(rs: RandomSource = RandomSource.Default): A = single(rs)
  */
 fun <A> Arb<A>.filter(predicate: (A) -> Boolean): Arb<A> = object : Arb<A>() {
    override fun edgecases(): List<A> = this@filter.edgecases().filter(predicate)
+
+   // sample the random edgecase generator 10 times and sift those values that satisfies the given predicate
+   // otherwise fallback to Arb's sample.
+   override fun edgecases(rs: RandomSource): List<A> =
+      generateSequence { this@filter.edgecases(rs) }
+         .take(10)
```

Comment:
I'm not convinced about this magic number. filter is kind of weird for this anyway, and iterating it multiple times is like expecting the random sample to somehow resolve something that is outside of the edgecases. maybe i should just take it once.

Reply:
Once or twice maybe yeah.
I think once you start with filters, edge cases starts to break down anyway.